sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -91,6 +97,27 @@ object PhysicalOperation extends PredicateHelper {
           .map(Alias(_, a.name)(a.exprId, a.qualifier)).getOrElse(a)
     }
   }
+
+  private def getDeterminedExpression(expr: Expression): Option[Expression] = {
```

Comment:
getDeterministicExpression or extractDeterministicExpression ? What do you think @maropu ?

Reply:
yea, I like the name including `Deterministic`